checkbox-ng/checkbox_ng/launcher/stages.py

```diff
@@ -132,13 +132,30 @@ def _run_single_job_with_ui_loop(self, job, ui):
                             comments += new_comment + '\n'
                         continue
                     elif cmd == 'skip':
-                        result_builder = JobResultBuilder(
-                            outcome=IJobResult.OUTCOME_SKIP,
-                            comments=_("Explicitly skipped before"
-                                       " execution"))
-                        if comments != "":
-                            result_builder.comments = comments
-                        break
+                        if (
```

Comment:
That piece of code is already a behemoth.
How difficult it would be to outline that function? I know it interacts with the for loop it's in, but the outlined function could returned something like "dealt_with" that could be used to continue/break.